tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Service/Microsoft.Legal.MatterCenter.Repository/Matter/MatterRepository.cs

```diff
@@ -1082,21 +1083,29 @@ public bool SetItemPermission(ClientContext clientContext, IList<IList<string>>
                                 {
                                     // Removing permission for all the user except current user with full control
                                     // Add those users in list, then traverse the list and removing all users from that list
-                                    
+
                                     //Get email from the user instead of name for comparison
                                     Principal principal = role.Member;
-                                    User currentListUser = (User)principal;
-                                    if (currentListUser.Email.ToLower() == user.ToLower() && !((role.Member.Title.ToLower() == loggedInUserTitle.ToLower()) && (roleDef.Name ==
-                                        matterSettings.EditMatterAllowedPermissionLevel)))
+                                    if (principal.GetType().Name != "Group")
```

Comment:
We need to check for the Group condition. Otherwise we will get type cast exception